src/librustc_resolve/diagnostics.rs

```diff
@@ -1223,17 +1197,12 @@ impl Foo for i32 {}
 }
 
 register_diagnostics! {
-//  E0153, unused error code
-//  E0157, unused error code
     E0254, // import conflicts with imported crate in this module
-//  E0257,
-//  E0258,
     E0402, // cannot use an outer type parameter in this context
     E0406, // undeclared associated type
-//  E0410, merged into 408
-    E0418, // is not an enum variant, struct or const
-    E0420, // is not an associated const
-    E0421, // unresolved associated const
+    E0418, // X bindings cannot shadow Ys
+    E0419, // unresolved pattern path kind `name`
+    E0420, // expected pattern path kind, found another pattern path kind
```

Comment:
If the error is technically the same as it was before (and fails for the same code) it is fine to reuse the code, otherwise a new code should be made.

Basically, if an explanation has to be rewritten to great extent to match the new semantics of the error code, a different code should be used.